src/services/db/__tests__/GitFileSystemService.spec.ts

```diff
@@ -677,182 +812,224 @@ describe("GitFileSystemService", () => {
+        "fake-user-id",
+        "fake content",
+        "fake-dir",
+        "create-file"
       )
 
       expect(actual._unsafeUnwrap()).toEqual(expected)
     })
 
-    it("should return a NotFoundError if the file does not exist", async () => {
+    it("should create a directory and a file if the directory doesn't already exist", async () => {
+      const expectedSha = "fake-hash"
       MockSimpleGit.cwd.mockReturnValueOnce({
-        revparse: jest.fn().mockRejectedValueOnce(new GitError()),
+        log: jest.fn().mockResolvedValueOnce({
+          latest: {
+            author_name: "fake-author",
+            author_email: "fake-email",
+            date: "fake-date",
+            message: "fake-message",
+            hash: "test-commit-sha",
+          },
+        }),
+      })
+      MockSimpleGit.cwd.mockReturnValueOnce({
+        checkIsRepo: jest.fn().mockResolvedValueOnce(true),
+      })
+      MockSimpleGit.cwd.mockReturnValueOnce({
+        remote: jest
+          .fn()
+          .mockResolvedValueOnce(
+            `git@github.com:${ISOMER_GITHUB_ORG_NAME}/fake-repo.git`
+          ),
+      })
+      MockSimpleGit.cwd.mockReturnValueOnce({
+        revparse: jest.fn().mockResolvedValueOnce("fake-hash"),
+      })
+      MockSimpleGit.cwd.mockReturnValueOnce({
+        checkout: jest.fn().mockResolvedValueOnce(undefined),
+      })
+      MockSimpleGit.cwd.mockReturnValueOnce({
+        add: jest.fn().mockReturnValueOnce({
+          commit: jest.fn().mockResolvedValueOnce({ commit: expectedSha }),
+        }),
       })
 
-      const result = await GitFileSystemService.read(
+      const expected = {
+        newSha: expectedSha,
+      }
+      const actual = await GitFileSystemService.create(
         "fake-repo",
-        "fake-dir/non-existent-file"
+        "fake-user-id",
+        "fake content",
+        "fake-create-dir",
+        "create-file"
       )
 
-      expect(result._unsafeUnwrapErr()).toBeInstanceOf(NotFoundError)
+      expect(actual._unsafeUnwrap()).toEqual(expected)
     })
 
-    it("should return a error if an error occurred when getting the Git blob hash", async () => {
+    it("should return an error if the file already exists", async () => {
       MockSimpleGit.cwd.mockReturnValueOnce({
-        revparse: jest.fn().mockRejectedValueOnce(new GitError()),
+        log: jest.fn().mockResolvedValueOnce({
+          latest: {
+            author_name: "fake-author",
+            author_email: "fake-email",
+            date: "fake-date",
+            message: "fake-message",
+            hash: "test-commit-sha",
+          },
+        }),
       })
-
-      const result = await GitFileSystemService.read(
+      const actual = await GitFileSystemService.create(
         "fake-repo",
-        "fake-dir/fake-file"
+        "fake-user-id",
+        "fake content",
+        "fake-dir",
+        "fake-file"
       )
 
-      expect(result.isErr()).toBeTrue()
+      expect(actual.isErr()).toBeTrue()
```

Comment:
we can assert the type of error returned with instanceOf

Reply:
Fixed in https://github.com/isomerpages/isomercms-backend/pull/892/commits/f0dd458b5ecea65092b685a34ec2e332d591f997